include/head/core/tasks/presence_sensing_driver_task.hpp

```diff
@@ -84,10 +101,13 @@ class PresenceSensingDriverTask {
         auto handler =
             PresenceSensingDriverMessageHandler{*driver, *can_client};
         TaskMessage message{};
+
         for (;;) {
             if (queue.try_read(&message, queue.max_delay)) {
                 handler.handle_message(message);
             }
+            handler.handle_message(
```

Comment:
What are we trying to achieve with the push tool notification? Are we wanting to check if a tool is present periodically? As this is written, the check for tools will only happen when a can message asking for voltage arrives. 

When we chatted about this the other day I suggested a timer sending `PushToolsDetectedNotification` to this task.

Reply:
Agree, and also it's not a good idea IMO to reuse messages like this. `can_messages::PushToolsDetectedNotification` is being used as 
a) something that the head sends via the canbus to the host to notify it of a change in the currently-attached tools
b) a head-internal message that's used as a trigger to actually do the check

That second one shouldn't be a canbus message. Also like Amit says we should use a timer or something for this.

Reply:
I've tried putting in something to get this bit taken care of using the FreeRTOSTimer. Trying to figure out a way of writing a test for it